### PR TITLE
Example of views and models refactoring

### PR DESCRIPTION
**Pull request by boonya**
Look at this PR and comment please

### CODE REVIEW

**Comment by boonya** [comment]
FK. I forgot to implement the tests (

File: anarchoApp/anarcho/helpers/response.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+class Response(object):
+    """
+    Response wrapper helper.
+    """
+
+    @staticmethod
+    def success(body, code=200):
+        """
+        Returns success ful response.
```
Comment (rmk135) [inline anarchoApp/anarcho/helpers/response.py]: successful

File: anarchoApp/anarcho/helpers/response.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    @staticmethod
+    def failure(reason, code=403):
+        """
+        Returns response with serialized json string about error reason.
+
+        :param str reason:
+        :param int code:
+        :return:
+        """
+        return make_response(json.dumps({
```
Comment (rmk135) [inline anarchoApp/anarcho/helpers/response.py]: JS Style line breaks

File: anarchoApp/anarcho/helpers/response.py (hunk 1 above)
Comment (boonya) [inline anarchoApp/anarcho/helpers/response.py]: Yep )

File: anarchoApp/anarcho/models/user.py [hunk 3]
```diff
@@ -26,11 +30,67 @@ def __init__(self, email=None, name=None, password=None):
...
+        Create new user or activate invited.
+
+        :param str name:
+        :param str email:
+        :param str password:
+        :return:
+        """
+
+        # Check name
+        if name.isspace():
```
Comment (rmk135) [inline anarchoApp/anarcho/models/user.py]: Separate method candidate

File: anarchoApp/anarcho/models/user.py [hunk 4]
```diff
@@ -26,11 +30,67 @@ def __init__(self, email=None, name=None, password=None):
...
+            raise ValueError(errors.USER_NAME_IS_EMPTY)
+        elif not 1 < len(name) < 20:
+            raise ValueError(errors.INVALID_USER_NAME_LENGTH)
+
+        # Check email
```
Comment (rmk135) [inline anarchoApp/anarcho/models/user.py]: Separate method candidate

File: anarchoApp/anarcho/models/user.py [hunk 5]
```diff
@@ -26,11 +30,67 @@ def __init__(self, email=None, name=None, password=None):
...
+        if email.isspace():
+            raise ValueError(errors.EMAIL_IS_EMPTY)
+        elif not 5 < len(email) < 50:
+            raise ValueError(errors.INVALID_EMAIL_LENGTH)
+        elif not re.search(r'\w[\w\.-]*@\w[\w\.-]+\.\w+', email):
+            raise ValueError(errors.INVALID_EMAIL_FORMAT)
+
+        # Check password
```
Comment (rmk135) [inline anarchoApp/anarcho/models/user.py]: Separate method candidate

File: anarchoApp/anarcho/models/user.py [hunk 6]
```diff
@@ -26,11 +30,67 @@ def __init__(self, email=None, name=None, password=None):
...
+        if password.isspace():
+            raise ValueError(errors.EMPTY_PASSWORD)
+        elif not 5 < len(password) < 100:
+            raise ValueError(errors.INVALID_PASSWORD_LENGTH)
+
+        email = email.lower()
+
+        # Find the same user
+        user = cls.query.filter(cls.email == email).first()
```
Comment (rmk135) [inline anarchoApp/anarcho/models/user.py]: Separate method candidate

File: anarchoApp/anarcho/routes/auth.py [hunk 7]
```diff
@@ -1,10 +1,11 @@
...
-from anarcho.models.token import Token
-from anarcho.serializer import serialize
 from anarcho.models.user import User
-from flask import request, g, Response, make_response
+from anarcho.serializer import serialize
+from anarcho.reasons import errors
+from anarcho.helpers.response import Response
 
 
 @app.route('/api/register', methods=['POST'])
```
Comment (rmk135) [inline anarchoApp/anarcho/routes/auth.py]: Definitely great refactor! Slim views are cool!

**Comment by rmk135** [comment]
It may be a little bit unexpected, but Python style guide that describes code commenting tells us to use such style for one line class comment:

```python
class SomeClass(object):

    """Some class."""
```

The following for one line function/method comment:

```python
def some_fn():
    """"Some function.""""
```

And, attention, such style for multiline comments of everything:

```python
def some_fn():
    """"Some function, short, one line description.
    
    Long description of some function with parameters definition and so on...
    """"
```

You can get more from PEP257
Also you can use https://pypi.python.org/pypi/pep257/0.5.0 for automated checks.

**Pull request closed without merging**